### DISCUSSION TITLE
Can I handle the onChange event of a controlled component?

### DISCUSSION THREAD

**Discussion by EmanueleCiriachi** [react-hook-form/react-hook-form#5995] [Q&A]
I have a Material UI TextField inside my form, connected to the form via a Controller, like so:

```
      <Controller
        control={props.control}
        {...props.register(props.fieldName)}
        render={({ field }) => (
          <TextField
            label={props.label}
            value={field.value}
            onChange={field.onChange}
            variant="outlined"
            type="number"
            InputLabelProps={props.inputLabelProps}
            inputProps={props.inputProps}
          />
        )}
      />
```

I would like to be able to filter its value via a regex, but the onChange event is tied to the Controller's field.
Can I manage the onChange event - filtering the values that will go into my TextField - without altering the regular functionality of react-hook-form?

**Comment by bluebill1049** [comment]
You can, just do a filter on the value before send to field.onChange
